File: task1.cpp
#include <iostream>
using namespace std;

double celsius_to_fahrenheit(double celsius) {
    return (celsius * 9.0 / 5.0) + 32;
}

double fahrenheit_to_celsius(double fahrenheit) {
    return (fahrenheit - 32) * 5.0 / 9.0;
}

double celsius_to_kelvin(double celsius) {
    return celsius + 273.15;
}

double kelvin_to_celsius(double kelvin) {
    return kelvin - 273.15;
}

double fahrenheit_to_kelvin(double fahrenheit) {
    return celsius_to_kelvin(fahrenheit_to_celsius(fahrenheit));
}

double kelvin_to_fahrenheit(double kelvin) {
    return celsius_to_fahrenheit(kelvin_to_celsius(kelvin));
}

int main() {
    char choice;
    double temperature;

    cout << "Enter the temperature: ";
    cin >> temperature;

    cout << "Enter the unit of temperature (C, F, or K): ";
    cin >> choice;

    switch(choice) {
        case 'C':
        case 'c':
            cout << temperature << " Celsius is equivalent to:" << endl;
            cout << celsius_to_fahrenheit(temperature) << " Fahrenheit" << endl;
            cout << celsius_to_kelvin(temperature) << " Kelvin" << endl;
            break;
        case 'F':
        case 'f':
            cout << temperature << " Fahrenheit is equivalent to:" << endl;
            cout << fahrenheit_to_celsius(temperature) << " Celsius" << endl;
            cout << fahrenheit_to_kelvin(temperature) << " Kelvin" << endl;
            break;
        case 'K':
        case 'k':
            cout << temperature << " Kelvin is equivalent to:" << endl;
            cout << kelvin_to_celsius(temperature) << " Celsius" << endl;
            cout << kelvin_to_fahrenheit(temperature) << " Fahrenheit" << endl;
            break;
        default:
            cout << "Invalid choice!" << endl;
    }

    return 0;
}

File: Task_1.py

import ipywidgets as widgets
from IPython.display import display

def convert_temperature(initial_temp, temp_value):
    try:
        if initial_temp == "Celsius":
            celsius = temp_value
            fahrenheit = (celsius * 9/5) + 32
            kelvin = celsius + 273.15
        elif initial_temp == "Fahrenheit":
            fahrenheit = temp_value
            celsius = (fahrenheit - 32) * 5/9
            kelvin = (fahrenheit - 32) * 5/9 + 273.15
        elif initial_temp == "Kelvin":
            kelvin = temp_value
            celsius = kelvin - 273.15
            fahrenheit = (kelvin - 273.15) * 9/5 + 32

        fahrenheit_label.value = f"{fahrenheit:.2f} °F"
        celsius_label.value = f"{celsius:.2f} °C"
        kelvin_label.value = f"{kelvin:.2f} K"
    except ValueError:
        fahrenheit_label.value = "Invalid input"
        celsius_label.value = "Invalid input"
        kelvin_label.value = "Invalid input"

# Initial Temperature Unit Dropdown
initial_temp_dropdown = widgets.Dropdown(
    options=["Celsius", "Fahrenheit", "Kelvin"],
    value="Celsius",
    description="Initial Temp Unit:"
)
display(initial_temp_dropdown)

# Temperature Value Entry
temp_value = widgets.FloatText(description="Temperature:")
display(temp_value)

# Fahrenheit Result
fahrenheit_label = widgets.Label(value="")
display(fahrenheit_label)

# Celsius Result
celsius_label = widgets.Label(value="")
display(celsius_label)

# Kelvin Result
kelvin_label = widgets.Label(value="")
display(kelvin_label)

# Convert Button
convert_button = widgets.Button(description="Convert")
display(convert_button)

def on_convert_button_clicked(b):
    convert_temperature(initial_temp_dropdown.value, temp_value.value)

convert_button.on_click(on_convert_button_clicked)
